src/instructlab/rag/rag_configuration.py

```diff
@@ -0,0 +1,79 @@
+# Standard
+import enum
+import os
+
+# Third Party
+from pydantic import BaseModel, ConfigDict, Field
+import click
+
+# First Party
+from instructlab.configuration import DEFAULTS
+
+
+class RagFramework(enum.Enum):
+    Haystack: str = "haystack"
+
+
+class DocumentStoreType(enum.Enum):
+    Milvus: str = "milvus"
+
+
+class DocumentStoreConfig(BaseModel):
+    # model configuration
+    model_config = ConfigDict(
+        extra="ignore",
+        protected_namespaces=(),
+        use_enum_values=False,  # populate models with the value property of enums, rather than the raw enum.
+    )
+    type: DocumentStoreType = Field(
+        default=DocumentStoreType.Milvus,
+        description=f"The document store type, one of: {[type.value for type in DocumentStoreType]}",
+    )
+    uri: str = Field(
+        default="embeddings.db",
+        description="The document store URI",
+    )
+    collection_name: str = Field(
+        default="IlabEmbeddings",
+        description="The document store collection name",
+    )
+
+
+class EmbeddingModelConfig(BaseModel):
+    # model configuration
+    model_config = ConfigDict(
+        extra="ignore",
+        protected_namespaces=(),
+        use_enum_values=True,  # populate models with the value property of enums, rather than the raw enum.
+    )
+    model_dir: str = Field(
+        default=DEFAULTS.MODELS_DIR,
+        description="The default system model location store, located in the data directory.",
+    )
+    model_name: str = Field(
+        # default_factory=lambda: "ibm-granite/granite-embedding-278m-multilingual",
```

Comment:
There's an ADR about model choice at https://github.com/instructlab/dev-docs/pull/169

Reply:
Updated accordingly to `ibm-granite/granite-embedding-125m-english`